### DISCUSSION TITLE
Kiali Operator won't launch Kiali instance due error: request() got an unexpected keyword argument 'chunked'

### DISCUSSION THREAD

**Discussion by joelcomp1** [kiali/kiali#6320] [Q&A]
I have tried this with both Kiali 1.65 and 1.68 and get the same issue so I suspect it's something on my end.  I am running K8s 1.27 on RKE2 and Kiali Operator is throwing this error constantly:

{"level":"error","ts":"2023-06-28T17:05:51Z","logger":"runner","msg":"Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/ansible_runner/__main__.py\", line 866, in main\n    res = run(**run_options)\n  File \"/usr/local/lib/python3.8/site-packages/ansible_runner/interface.py\", line 260, in run\n    r.run()\n  File \"/usr/local/lib/python3.8/site-packages/ansible_runner/runner.py\", line 119, in run\n    self.status_callback('starting')\n  File \"/usr/local/lib/python3.8/site-packages/ansible_runner/runner.py\", line 107, in status_callback\n    ansible_runner.plugins[plugin].status_handler(self.config, status_data)\n  File \"/usr/local/lib/python3.8/site-packages/ansible_runner_http/events.py\", line 35, in status_handler\n    status = send_request(plugin_config['runner_url'],\n  File \"/usr/local/lib/python3.8/site-packages/ansible_runner_http/events.py\", line 18, in send_request\n    return session.post(url_actual, headers=headers, json=(data))\n  File \"/usr/local/lib/python3.8/site-packages/requests/sessions.py\", line 635, in post\n    return self.request(\"POST\", url, data=data, json=json, **kwargs)\n  File \"/usr/local/lib/python3.8/site-packages/requests/sessions.py\", line 587, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"/usr/local/lib/python3.8/site-packages/requests/sessions.py\", line 701, in send\n    r = adapter.send(request, **kwargs)\n  File \"/usr/local/lib/python3.8/site-packages/requests/adapters.py\", line 486, in send\n    resp = conn.urlopen(\n  File \"/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py\", line 790, in urlopen\n    response = self._make_request(\n  File \"/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py\", line 496, in _make_request\n    conn.request(\nTypeError: request() got an unexpected keyword argument 'chunked'\n","job":"2238350836631459930","name":"installer","namespace":"","error":"exit status 1","stacktrace":"github.com/operator-framework/operator-sdk/internal/ansible/runner.(*runner).Run.func1\n\t/workspace/internal/ansible/runner/runner.go:269"}
{"level":"error","ts":"2023-06-28T17:05:52Z","logger":"reconciler","msg":"Unable to mark error to run reconciliation","job":"2238350836631459930","name":"installer","namespace":"","error":"namespaces \"installer\" is forbidden: User \"system:serviceaccount:kiali:kiali-kiali-kiali-operator\" cannot update resource \"namespaces/status\" in API group \"\" in the namespace \"installer\"","stacktrace":"github.com/operator-framework/operator-sdk/internal/ansible/controller.(*AnsibleOperatorReconciler).Reconcile\n\t/workspace/internal/ansible/controller/reconcile.go:224\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:122\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:323\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:235"}

**Comment by jmazzitelli** [comment]
```
namespaces "installer" is forbidden: User "system:serviceaccount:kiali:kiali-kiali-kiali-operator" cannot update resource "namespaces/status" in API group "" in the namespace "installer""
```

Sounds like the operator's SA is incorrect and does not have the proper permissions.

How did you install the operator? Did you use the published Kiali helm charts, OLM, or some custom method?

**Comment by joelcomp1** [comment]
Hmm its a helm chart based on the Kiali helm charts, i checked the Cluster Role and it seems to match the latest on github:

`apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: kiali-kiali-kiali-operator
  labels:
    app: kiali-operator
    app.kubernetes.io/instance: kiali-kiali
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: kiali-operator
    app.kubernetes.io/part-of: kiali-operator
  annotations:
    meta.helm.sh/release-name: kiali-kiali
    meta.helm.sh/release-namespace: kiali
rules:
  - verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - ''
    resources:
      - configmaps
      - endpoints
      - pods
      - serviceaccounts
      - services
      - services/finalizers
  - verbs:
      - get
      - list
      - patch
    apiGroups:
      - ''
    resources:
      - namespaces
  - verbs:
      - create
      - list
      - watch
    apiGroups:
      - ''
    resources:
      - secrets
  - verbs:
      - get
    apiGroups:
      - ''
    resources:
      - secrets
    resourceNames:
      - cacerts
      - istio-ca-secret
  - verbs:
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - ''
    resources:
      - secrets
    resourceNames:
      - kiali-signing-key
  - verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - apps
    resources:
      - deployments
      - replicasets
  - verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - autoscaling
    resources:
      - horizontalpodautoscalers
  - verbs:
      - create
      - get
    apiGroups:
      - monitoring.coreos.com
    resources:
      - servicemonitors
  - verbs:
      - update
    apiGroups:
      - apps
    resources:
      - deployments/finalizers
    resourceNames:
      - kiali-operator
  - verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - kiali.io
    resources:
      - '*'
  - verbs:
      - list
    apiGroups:
      - authorization.k8s.io
    resources:
      - selfsubjectaccessreviews
  - verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - rbac.authorization.k8s.io
    resources:
      - clusterrolebindings
      - clusterroles
      - rolebindings
      - roles
  - verbs:
      - get
      - list
      - watch
    apiGroups:
      - apiextensions.k8s.io
    resources:
      - customresourcedefinitions
  - verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - extensions
      - networking.k8s.io
    resources:
      - ingresses
  - verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - route.openshift.io
    resources:
      - routes
  - verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - oauth.openshift.io
    resources:
      - oauthclients
  - verbs:
      - list
      - watch
    apiGroups:
      - config.openshift.io
    resources:
      - clusteroperators
  - verbs:
      - get
    apiGroups:
      - config.openshift.io
    resources:
      - clusteroperators
    resourceNames:
      - kube-apiserver
  - verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
    apiGroups:
      - console.openshift.io
    resources:
      - consolelinks
  - verbs:
      - get
      - list
      - watch
    apiGroups:
      - ''
    resources:
      - configmaps
      - endpoints
      - pods/log
  - verbs:
      - get
      - list
      - watch
      - patch
    apiGroups:
      - ''
    resources:
      - namespaces
      - pods
      - replicationcontrollers
      - services
  - verbs:
      - create
      - post
    apiGroups:
      - ''
    resources:
      - pods/portforward
  - verbs:
      - get
      - list
      - watch
      - patch
    apiGroups:
      - extensions
      - apps
    resources:
      - daemonsets
      - deployments
      - replicasets
      - statefulsets
  - verbs:
      - get
      - list
      - watch
      - patch
    apiGroups:
      - batch
    resources:
      - cronjobs
      - jobs
  - verbs:
      - get
      - list
      - watch
      - create
      - delete
      - patch
    apiGroups:
      - config.istio.io
      - networking.istio.io
      - authentication.istio.io
      - rbac.istio.io
      - security.istio.io
      - extensions.istio.io
      - telemetry.istio.io
      - gateway.networking.k8s.io
    resources:
      - '*'
  - verbs:
      - get
      - list
      - watch
      - patch
    apiGroups:
      - apps.openshift.io
    resources:
      - deploymentconfigs
  - verbs:
      - get
    apiGroups:
      - project.openshift.io
    resources:
      - projects
  - verbs:
      - get
    apiGroups:
      - route.openshift.io
    resources:
      - routes
  - verbs:
      - create
    apiGroups:
      - authentication.k8s.io
    resources:
      - tokenreviews
`


I even tried to all all verbs "*" for namespace but no dice same issue

**Comment by jmazzitelli** [comment]
Let me guess... Rancher? :)

Try an experiment - install the operator using the Kiali-provided helm charts: 

https://kiali.io/docs/installation/installation-guide/install-with-helm/#adding-the-kiali-helm-charts-repository

If you don't get the same error and Kiali server installs fine, then the problem is in the helm charts you are using.

If you get the same error, I would wonder if the cluster has some locked down security. I actually don't know what "status" permission it would need on the namespace - that's a new one to me.

**Comment by joelcomp1** [comment]
Ha no good guess though, actually from Platform One, it had worked before so I had thought it was maybe a k8s version thing as I had moved from 1.25 to 1.27 but that doesnt seem to be it.  I used the helm chart from the link you provided and worked great, i'll dig into it more on my side, thanks for the help!

**Comment by jmazzitelli** [comment]
_converted to Q&A Discussion since this isn't a bug issue_
